_posts/2021-01-12-github-application-running-on-faasd-behind-inlets.md

```diff
@@ -0,0 +1,508 @@
+![github-app](/images/2021-01-12-github-application-using-go-and-inlets-pro/app.png)
+  
+## Setup faasd
+In order to get up and running with your own faasd installation on your Mac you can use multipass.
+
+Let's start our Ubuntu VM with multipass.First, we need a cloud-config.txt to set up faasd while bootstrapping VM
+
+```bash
+$ curl -sSLO https://raw.githubusercontent.com/openfaas/faasd/master/cloud-config.txt
+```
+
+Then, we need to update the SSH key to match your own, edit cloud-config.txt:
+```
+$ ssh-keygen -t rsa -b 4096 -C "developerguyn@gmail.com" -f $PWD/id_rsa
+```
+
+Replace the _ssh_authorized_keys::ssh-rsa_ value with the contents of `~/.ssh/id_rsa.pub`, which is defined in `cloud-config.txt`.
+
+Finally, boot the VM
+
+```bash
+$ multipass launch --cloud-init cloud-config.txt  --name faasd
+```
+
+Check the VM if it is working properly
+
+```bash
+$ multipass list                                                                                                                                                                                             🐳(docker-desktop)
+Name                    State             IPv4             Image
+faasd                   Running           192.168.64.25    Ubuntu 20.04 LTS
+```
+
+For more details you can check it out the [link](https://github.com/openfaas/faasd/blob/master/docs/MULTIPASS.md).
+
+## Build&Deploy function
+In this demo, we are going to use Go to develop our Github Application, in order to do that, first, we need to pull the corresponding function template for the Go.
+
+```bash
+# let's look at the available Go function templates within the OpenFaaS store
+$ faas-cli template store list | grep -i "go"
+go                       openfaas           Classic Golang template
+golang-http              openfaas           Golang HTTP template
+golang-middleware        openfaas           Golang Middleware template
+# We are going to use golang-middleware function template, let's pull it.
+$ faas-cli template store pull golang-middleware
+# Then, create the function itself.
+$ faas-cli new issues-bot --lang golang-middleware --prefix <DOCKER_HUB_ID>
+```
+After created the function, we need to define some arguments, environments and secrets for the function.
+Let's add them:
+
+```yaml
+  build_args:
+      GO111MODULE: on
+    secrets:
+      - webhook-secret # your secret goes here
+      - private-key-secret # your private key goes here
+    environment:
+      APP_ID: "" #your app id goes here
+```
+
+Finally, we need to create those secrets above with make use of faas-cli.
+
+Let's create our secrets
+
+```bash
+$ export WEBHOOK_SECRET="sup3rs3cr3t"
+$ faas-cli secret create webhook-secret --from-literal $WEBHOOK_SECRET
+# Download the private key to your host
+$ faas-cli secret create private-key-secret --from-file <path_to_your_pem_file>.pem
+```
+
+We should create a secret in faasd, in order to do that we need to access the Gateway of faasd
```

Comment:
Also mention that they can run `multipass info faasd ` at any time and show the output it gives in long form please.

Reply:
done